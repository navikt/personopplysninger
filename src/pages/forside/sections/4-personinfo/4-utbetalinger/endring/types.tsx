import { OptionType } from '../../../../../../types/option';

export interface OutboundNorskKontonummer {
    value: string;
}

export interface OutboundUtenlandsbankonto {
    value: string | null;
    utenlandskKontoInformasjon: {
        landkode: string;
        landkodeTobokstavs?: string;
        valuta: string;
        swift?: string;
        bank: {
            adresseLinje1?: string;
            adresseLinje2?: string;
            adresseLinje3?: string;
            kode?: string;
            navn: string;
        };
    };
}

export interface FormFields {
    kontonummer: string;
    land?: OptionType;
    valuta?: OptionType;
    banknavn: string;
    kontonummerIban: string;
    bickode: string;
    retningsnummer: string;
    bankkode: string;
    adresse1: string;
    adresse2: string;
    adresse3: string;
}
