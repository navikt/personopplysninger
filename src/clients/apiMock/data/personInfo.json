{
  "personalia": {
    "fornavn": "DONALD FAUNTLEROY",
    "etternavn": "DUCK",
    "personident": {
      "verdi": "12345678911",
      "type": "FNR"
    },
    "utenlandskbank": {
      "adresse1": "Gävlegatan 22",
      "adresse2": "113 30 Stockholm",
      "adresse3": "",
      "bankkode": "90081210188",
      "banknavn": "Nordax Bank AB",
      "iban": "",
      "kontonummer": "US90081210188",
      "land": "Sverige",
      "swiftkode": "HELGSE21",
      "valuta": "Tugriks Mongolia"
    },
    "tlfnr": {
      "jobb": "12121212",
      "mobil": "24242424",
      "privat": "36363636",
      "datakilder": [{}, {}]
    },
    "spraak": "NB",
    "epostadr": "TODO",
    "personstatus": "BOSA",
    "statsborgerskap": "NOR",
    "foedested": "NOR",
    "sivilstand": "UGIF",
    "kjoenn": "M",
    "datakilder": [{}, {}, {}, {}]
  },
  "enhetKontaktInformasjon": {
    "enhet": {
      "postadresse": {
        "type": "stedsadresse",
        "postnummer": "0614",
        "poststed": "Oslo",
        "gatenavn": "Slottsveien",
        "husnummer": "22",
        "husbokstav": "A"
      },
      "publikumsmottak": [
        {
          "gateadresse": "Peder Falcks veg",
          "poststed": "Trondheim",
          "husnummer": "31",
          "husbokstav": null,
          "postnummer": "7044",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        },
        {
          "gateadresse": "Kirkegata",
          "poststed": "Trondheim",
          "husnummer": "25",
          "husbokstav": null,
          "postnummer": "7014",
          "stedsbeskrivelse": null,
          "aapningMandag": {
            "dag": "Mandag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": "Timeavtaler også utenom åpningstiden"
          },
          "aapningTirsdag": {
            "dag": "Tirsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningOnsdag": {
            "dag": "Onsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningTorsdag": {
            "dag": "Torsdag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningFredag": {
            "dag": "Fredag",
            "fra": "10:00",
            "til": "14:00",
            "stengt": "false",
            "kommentar": null
          },
          "aapningAndre": [
            {
              "dag": "15.05.2019",
              "fra": "09:00",
              "til": "12:00",
              "stengt": "false",
              "kommentar": "Her kan det også være en merknad "
            },
            {
              "dag": "17.05.2019",
              "fra": null,
              "til": null,
              "stengt": "true"
            },
            {
              "dag": "29.05.2019",
              "fra": "09:00",
              "til": "13:30",
              "stengt": "false",
              "kommentar": "Onsdag før Kristi himmelfartsdag"
            }
          ]
        }
      ],
      "tlfperson": "55553333",
      "spesielleopplysninger": "Timeavtale 8-15, man - fre."
    }
  },
  "adresser": {
    "boadresse": {
      "adressetillegg": "C/O Rune Aremark",
      "adresse": "Bjørvika 4D",
      "datoFraOgMed": "2017-02-21",
      "kommune": "1239",
      "land": "Norge",
      "matrikkeladresse": {},
      "postnummer": "1239",
      "poststed": "Jepp",
      "veiadresse": {
        "bokstav": "A",
        "bolignummer": "H0102",
        "gatekode": "01239",
        "husnummer": "11"
      }
    },
    "geografiskTilknytning": {
      "bydel": "500102",
      "datoFraOgMed": "2012-10-01",
      "kommune": null,
      "land": null,
      "enhet": "NAV Falkenborg"
    },
    "postadresse": {
      "adresse1": "12345 HFJEF IQUFJ HFURHGHQBEUQ",
      "adresse2": "BATOOM LUNCHAGRAMA,GONGRUPEA,DD",
      "adresse3": "GGLAM,10000 HCAMGHU,THAILAND",
      "datoFraOgMed": "2018-03-13",
      "land": "THA",
      "postnummer": "",
      "poststed": "Nittedal"
    },
    "prioritertAdresse": "TIAD",
    "tilleggsadresse": {
      "adresse1": "JEPP 10jfioew JIO LPFL ",
      "adresse2": "LOOOL FKEOWIF JEWIOF WECOOOO FJEI",
      "adresse3": "",
      "bolignummer": "",
      "bydel": "",
      "datoFraOgMed": "2019-01-12",
      "datoTilOgMed": "2023-01-11",
      "gateKode": "0270",
      "husbokstav": "",
      "husnummer": "0020",
      "kommunenummer": "0131",
      "postboksanlegg": "",
      "postboksnummer": "",
      "postnummer": "2021",
      "poststed": "JAJAJA"
    },
    "datakilder": [{}, {}]
  }
}
