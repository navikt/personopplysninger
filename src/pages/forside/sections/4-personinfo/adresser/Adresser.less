.addresse {
  &__box {
    .addresse__lesmer {
      .lesMerPanel__toggle {
        justify-content: flex-start;
      }
    }
  }
  &__tittel {
    padding: 0 0 0.5rem 0;
  }
  &__lesmer {
    padding: 0;
  }
  &__linje {
    width: 100%;
  }
}
