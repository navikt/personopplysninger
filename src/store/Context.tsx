import { createContext, useContext, useReducer, Dispatch } from 'react';
import { Action, initialState, reducer, Store } from './Store';
interface Props {
    children: JSX.Element | JSX.Element[];
}

export const StoreContext = createContext({} as [Store, Dispatch<Action>]);
export const StoreProvider = (props: Props) => {
    const { children } = props;

    return <StoreContext.Provider value={useReducer(reducer, initialState)}>{children}</StoreContext.Provider>;
};
export const useStore = () => useContext(StoreContext);
