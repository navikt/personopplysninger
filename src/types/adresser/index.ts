import { Boadresse } from "./boadresse";
import { Postadresse } from "./postadresse";
import { Tilleggsadresse } from "./tilleggsadresse";
import { UtenlandskAdresse } from "./utenlandskadresse";

export interface GeografiskTilknytning {
  bydel?: string;
  datoFraOgMed?: string;
  kommune?: string;
  enhet?: string;
  land?: string;
}

export interface Adresser {
  boadresse?: Boadresse;
  geografiskTilknytning?: GeografiskTilknytning;
  postadresse?: Postadresse;
  prioritertAdresse?: string;
  tilleggsadresse?: Tilleggsadresse;
  utenlandskAdresse?: UtenlandskAdresse;
  datakilder?: {}[];
}
