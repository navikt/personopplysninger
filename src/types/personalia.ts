export interface Tlfnr {
    telefonHoved?: string;
    telefonAlternativ?: string;
    landskodeHoved?: string;
    landskodeAlternativ?: string;
}

export interface UtenlandskBankkonto {
    adresse1: string;
    adresse2: string;
    adresse3: string;
    bankkode: string;
    banknavn: string;
    iban: string;
    kontonummer: string;
    land: string;
    swiftkode: string;
    valuta: string;
}

export interface Personalia {
    fornavn?: string;
    etternavn?: string;
    personident?: {
        verdi: string;
        type: string;
    };
    kontoregisterStatus: string;
    kontonr?: string;
    utenlandskbank: UtenlandskBankkonto;
    tlfnr?: Tlfnr;
    statsborgerskap?: string[];
    foedested?: string;
    sivilstand?: string;
    kjoenn?: string;
}
