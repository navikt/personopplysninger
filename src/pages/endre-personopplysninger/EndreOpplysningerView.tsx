import AdresserVisning from '@/pages/forside/sections/4-personinfo/3-adresser/Adresser';
import Utbetalinger from '@/pages/forside/sections/4-personinfo/4-utbetalinger/Utbetalinger';
import Box from '@/components/box/Box';
import kontaktIkon from '@/assets/img/Kontakt.svg';
import TelefonnummerHosNav from '@/pages/forside/sections/4-personinfo/2-kontaktinfo/subsections/telefonnummer/TelefonnummerHosNav';
import { Personalia } from '../../types/personalia';
import { Adresser } from '../../types/adresser';

interface Props {
    personalia?: Personalia;
    adresser?: Adresser;
}

const EndreOpplysningerView = (props: Props) => {
    const { personalia, adresser } = props;
    return (
        <>
            {personalia && (
                <Box id="kontaktinformasjon" tittel="kontaktinfo.tittel" icon={kontaktIkon}>
                    <TelefonnummerHosNav tlfnr={personalia.tlfnr} />
                </Box>
            )}
            {adresser && <AdresserVisning adresser={adresser} />}
            {personalia && (
                <Utbetalinger
                    kontonr={personalia.kontonr}
                    utenlandskbank={personalia.utenlandskbank}
                    kontoregisterStatus={personalia.kontoregisterStatus}
                />
            )}
        </>
    );
};

export default EndreOpplysningerView;
