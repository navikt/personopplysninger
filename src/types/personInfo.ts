import { Personalia } from "./personalia";
import { Adresser } from "./adresser";
import { EnhetKontaktInfo } from "./enhetKontaktInfo";

export interface PersonInfo {
  personalia?: Personalia;
  adresser?: Adresser;
  enhetKontaktInformasjon?: EnhetKontaktInfo;
}
