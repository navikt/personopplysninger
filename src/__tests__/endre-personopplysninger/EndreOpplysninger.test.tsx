import '@testing-library/jest-dom';
import personInfo from '../../clients/apiMock/app/fetch/person-info.json';
import { IntlProvider } from 'react-intl';
import nbMessages from 'text/nb';
import EndreOpplysningerView from '../../pages/endre-personopplysninger/EndreOpplysningerView';
import { Adresser } from '../../types/adresser';
import { Personalia } from '../../types/personalia';
import { StoreProvider } from '@/store/Context';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-modal');

describe('EndreOpplysningerView', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <StoreProvider>
                <IntlProvider locale={'nb'} messages={nbMessages}>
                    <MemoryRouter>
                        <EndreOpplysningerView
                            adresser={personInfo.adresser as unknown as Adresser}
                            personalia={personInfo.personalia as unknown as Personalia}
                        />
                    </MemoryRouter>
                </IntlProvider>
            </StoreProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
