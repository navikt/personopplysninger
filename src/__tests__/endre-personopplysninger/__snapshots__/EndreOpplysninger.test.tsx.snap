// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndreOpplysningerView renders correctly 1`] = `
<DocumentFragment>
  <div
    class="box__wrapper"
    id="kontaktinformasjon"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <div
          class="navds-guide__illustration navds-guide__illustration--medium"
        >
          <img
            alt=""
            class="box__ikon"
            src="Kontakt.svg"
          />
        </div>
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Kontaktinformasjon
              </h2>
              <div
                class="box__line"
              />
            </div>
          </div>
          <div
            class="box__content"
          >
            <div>
              <div
                class="tlfnummer__rad"
              >
                <div
                  class="tlfnummer__container"
                >
                  <div
                    class="tlfnummer__verdi"
                  >
                    <p
                      class="navds-label"
                    >
                      Telefon 
                    </p>
                    <p
                      class="navds-body-short"
                    >
                      <span>
                        +47 
                      </span>
                      408 53 158
                    </p>
                  </div>
                  <div
                    class="tlfnummer__knapper"
                  >
                    <button
                      aria-label="Endre telefonnummer"
                      class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
                    >
                      <span
                        aria-live="polite"
                        class="navds-label"
                      >
                        <img
                          alt=""
                          class="tlfnummer__knapp-ikon"
                          src="Pencil.svg"
                        />
                        <span
                          class="tlfnummer__knapp-tekst"
                        >
                          Endre
                        </span>
                      </span>
                    </button>
                    <button
                      aria-label="Slett telefonnummer"
                      class="knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
                    >
                      <span
                        aria-live="polite"
                        class="navds-label"
                      >
                        <img
                          alt=""
                          class="tlfnummer__knapp-ikon"
                          src="Slett.svg"
                        />
                        <span
                          class="tlfnummer__knapp-tekst"
                        >
                          Slett
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <button
              aria-label="Legg til telefonnummer i NAV sitt register"
              class="tlfnummer__leggtil knapp-med-ikon lenke navds-button navds-button--tertiary navds-button--medium"
            >
              <span
                aria-live="polite"
                class="navds-label"
              >
                <img
                  alt=""
                  class="kilde__icon"
                  src="LeggTil.svg"
                />
                Legg til nummer i NAV sitt register
              </span>
            </button>
            <div
              class="kilde__container"
            >
              <div
                class="kilde__tekst"
              >
                <p
                  class="navds-detail"
                >
                  Kilde: NAV
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="box__wrapper"
    id="adresser"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <div
          class="navds-guide__illustration navds-guide__illustration--medium"
        >
          <img
            alt=""
            class="box__ikon"
            src="Adresse.svg"
          />
        </div>
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Adresser
              </h2>
              <button
                aria-label="Les mer om denne seksjonen."
                aria-pressed="false"
                class="infotekst__title-i-button"
              >
                <img
                  alt=""
                  class="infotekst__title-i-icon"
                  src="Informasjonstekst.svg"
                />
              </button>
              <div
                class="box__line"
              />
            </div>
            <a
              class="navds-link anchor-link"
              href="#adresser"
            >
              <img
                alt=""
                src="AnchorLink.svg"
              />
              Lenke hit
            </a>
          </div>
          <div
            class="box__content"
          >
            <div
              class="adresse__box"
            >
              <div>
                <div
                  class="underseksjon__header"
                >
                  <h3
                    class="navds-heading navds-heading--small"
                  >
                    Adresser som er registert i Folkeregisteret
                  </h3>
                </div>
                <div
                  class="adresse__box"
                >
                  <div
                    class="underseksjon__header"
                  >
                    <h4
                      class="navds-heading navds-heading--xsmall"
                    >
                      Bostedsadresse
                    </h4>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      V/ Aremark Jr
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      JÃ˜RGENSENVEIEN 14 B
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      0194 OSLO
                    </p>
                  </div>
                  <div
                    class="navds-read-more navds-read-more--medium adresse__lesmer"
                  >
                    <button
                      aria-expanded="false"
                      class="navds-read-more__button navds-body-short"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="navds-read-more__expand-icon"
                        fill="none"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="navds-read-more__expand-icon navds-read-more__expand-icon--filled"
                        fill="none"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="m12 18.121-8.06-8.06 2.12-2.122L12 13.88l5.94-5.94 2.12 2.122L12 18.12Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                      <span>
                        Vis detaljer
                      </span>
                    </button>
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        aria-hidden="true"
                        style="display: none;"
                      >
                        <div
                          class="navds-read-more__content navds-body-long"
                        >
                          <dl
                            class="list address-columns"
                          >
                            <div
                              class="list-item"
                            >
                              <dt
                                class="list-element-heading"
                              >
                                <span
                                  class="navds-label"
                                >
                                  Bolignummer
                                </span>
                              </dt>
                              <dd>
                                H0301
                              </dd>
                            </div>
                            <div
                              class="list-item"
                            >
                              <dt
                                class="list-element-heading"
                              >
                                <span
                                  class="navds-label"
                                >
                                  Kommune
                                </span>
                              </dt>
                              <dd>
                                EVJE OG HORNNES
                              </dd>
                            </div>
                            <div
                              class="list-item"
                            >
                              <dt
                                class="list-element-heading"
                              >
                                <span
                                  class="navds-label"
                                >
                                  Flyttedato
                                </span>
                              </dt>
                              <dd>
                                14.07.2020
                              </dd>
                            </div>
                          </dl>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="adresse__divider"
                  />
                </div>
                <div
                  class="adresse__box"
                >
                  <div
                    class="underseksjon__header"
                  >
                    <h4
                      class="navds-heading navds-heading--xsmall"
                    >
                      Delt bosted
                    </h4>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      V/ Aremark Jr
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      JÃ˜RGENSENVEIEN 14 B
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      0194 OSLO
                    </p>
                  </div>
                  <div
                    class="navds-read-more navds-read-more--medium adresse__lesmer"
                  >
                    <button
                      aria-expanded="false"
                      class="navds-read-more__button navds-body-short"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="navds-read-more__expand-icon"
                        fill="none"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="navds-read-more__expand-icon navds-read-more__expand-icon--filled"
                        fill="none"
                        focusable="false"
                        height="1em"
                        role="img"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="m12 18.121-8.06-8.06 2.12-2.122L12 13.88l5.94-5.94 2.12 2.122L12 18.12Z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                      <span>
                        Vis detaljer
                      </span>
                    </button>
                    <div
                      style="height: 0px; overflow: hidden;"
                    >
                      <div
                        aria-hidden="true"
                        style="display: none;"
                      >
                        <div
                          class="navds-read-more__content navds-body-long"
                        >
                          <dl
                            class="list address-columns"
                          >
                            <div
                              class="list-item"
                            >
                              <dt
                                class="list-element-heading"
                              >
                                <span
                                  class="navds-label"
                                >
                                  Bolignummer
                                </span>
                              </dt>
                              <dd>
                                H0301
                              </dd>
                            </div>
                            <div
                              class="list-item"
                            >
                              <dt
                                class="list-element-heading"
                              >
                                <span
                                  class="navds-label"
                                >
                                  Kommune
                                </span>
                              </dt>
                              <dd>
                                EVJE OG HORNNES
                              </dd>
                            </div>
                          </dl>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="adresse__divider"
                  />
                </div>
                <div
                  class="adresse__box"
                >
                  <div
                    class="underseksjon__header"
                  >
                    <h4
                      class="navds-heading navds-heading--xsmall"
                    >
                      Oppholdsadresse
                    </h4>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      V/ Aremark Jr
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      JÃ˜RGENSENVEIEN 14 B
                    </p>
                  </div>
                  <div
                    class="adresse__linje"
                  >
                    <p
                      class="navds-body-short"
                    >
                      0194 OSLO
                    </p>
                  </div>
                  <div
                    class="adresse__divider"
                  />
                </div>
                <div
                  class="kilde__container"
                >
                  <p
                    class="navds-body-long"
                  >
                    <a
                      class="navds-link kilde__lenke lenke"
                      href="https://www.skatteetaten.no/person/folkeregister/flytte/"
                    >
                      <span
                        class="kilde__icon"
                      >
                        <img
                          alt=""
                          src="Link.svg"
                        />
                      </span>
                      Endre adresse i Folkeregisteret
                    </a>
                  </p>
                  <div
                    class="kilde__tekst"
                  >
                    <p
                      class="navds-detail"
                    >
                      Kilde: Folkeregisteret
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="underseksjon__header underseksjon__divider"
              >
                <h3
                  class="navds-heading navds-heading--small"
                >
                  Ã˜vrige registrerte adresser
                </h3>
              </div>
              <div
                class="adresse__box"
              >
                <div
                  class="underseksjon__header"
                >
                  <h4
                    class="navds-heading navds-heading--xsmall"
                  >
                    Kontaktadresse registrert hos NAV
                  </h4>
                </div>
                <div
                  class="adresse__linje"
                >
                  <p
                    class="navds-body-short"
                  >
                    V/ Aremark Jr
                  </p>
                </div>
                <div
                  class="adresse__linje"
                >
                  <p
                    class="navds-body-short"
                  >
                    Postboks 10 SÃ¸rstrand
                  </p>
                </div>
                <div
                  class="adresse__linje"
                >
                  <p
                    class="navds-body-short"
                  >
                    0194 OSLO
                  </p>
                </div>
                <div
                  class="navds-read-more navds-read-more--medium adresse__lesmer"
                >
                  <button
                    aria-expanded="false"
                    class="navds-read-more__button navds-body-short"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="navds-read-more__expand-icon"
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="img"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="navds-read-more__expand-icon navds-read-more__expand-icon--filled"
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="img"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="m12 18.121-8.06-8.06 2.12-2.122L12 13.88l5.94-5.94 2.12 2.122L12 18.12Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                    <span>
                      Vis detaljer
                    </span>
                  </button>
                  <div
                    style="height: 0px; overflow: hidden;"
                  >
                    <div
                      aria-hidden="true"
                      style="display: none;"
                    >
                      <div
                        class="navds-read-more__content navds-body-long"
                      >
                        <dl
                          class="list address-columns"
                        >
                          <div
                            class="list-item"
                          >
                            <dt
                              class="list-element-heading"
                            >
                              <span
                                class="navds-label"
                              >
                                Gyldig til og med
                              </span>
                            </dt>
                            <dd>
                              20.07.2020
                            </dd>
                          </div>
                        </dl>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="adresse__divider"
                />
              </div>
              <button
                class="adresse__slett-kontaktadresse knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
              >
                <span
                  class="navds-button__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10 3a1 1 0 0 0-1 1v1h6V4a1 1 0 0 0-1-1h-4Zm7 2V4a3 3 0 0 0-3-3h-4a3 3 0 0 0-3 3v1H2v2h2.064l.814 12.2A3 3 0 0 0 7.87 22h8.258a3 3 0 0 0 2.993-2.8L19.936 7H22V5h-5Zm.931 2H6.07l.804 12.067a1 1 0 0 0 .998.933h8.258a1 1 0 0 0 .998-.933L17.93 7ZM9 18V9h2v9H9Zm4 0V9h2v9h-2Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  aria-live="polite"
                  class="navds-label"
                >
                  Slett kontaktadresse
                </span>
              </button>
              <div
                class="kilde__container"
              >
                <div
                  class="kilde__tekst"
                >
                  <p
                    class="navds-detail"
                  >
                    Kilde: NAV
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="box__wrapper"
    id="utbetaling"
  >
    <div
      class="navds-guide-panel navds-guide-panel--poster"
    >
      <div
        class="navds-guide"
      >
        <div
          class="navds-guide__illustration navds-guide__illustration--medium"
        >
          <img
            alt=""
            class="box__ikon"
            src="Kontonummer.svg"
          />
        </div>
      </div>
      <div
        class="navds-guide-panel__content"
      >
        <div
          class="box__container"
        >
          <div
            class="box__header"
          >
            <div
              class="box__title-container"
            >
              <div
                class="box__line"
              />
              <h2
                class="box__title navds-heading navds-heading--medium"
              >
                Kontonummer for utbetaling
              </h2>
              <div
                class="box__line"
              />
            </div>
            <a
              class="navds-link anchor-link"
              href="#utbetaling"
            >
              <img
                alt=""
                src="AnchorLink.svg"
              />
              Lenke hit
            </a>
          </div>
          <div
            class="box__content"
          >
            <dl
              class="list"
            >
              <div
                class="list-item"
              >
                <dt
                  class="list-element-heading"
                >
                  <span
                    class="navds-label"
                  >
                    Kontonummer
                  </span>
                </dt>
                <dd>
                  1234 56 78911
                </dd>
              </div>
            </dl>
            <div
              class="kilde__container"
            >
              <button
                class="kilde__knapp knapp-med-ikon navds-button navds-button--tertiary navds-button--medium"
              >
                <span
                  aria-live="polite"
                  class="navds-label"
                >
                  <img
                    alt=""
                    class="kilde__icon"
                    src="Pencil.svg"
                  />
                  Endre
                </span>
              </button>
              <div
                class="kilde__tekst"
              >
                <p
                  class="navds-detail"
                >
                  Kilde: NAV
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
